### PR TITLE
Migrate VIT [WIP]

### LINKED ISSUE #1628: Migrate VIT to the new `Backbone` API
To align KerasCV with the KerasNLP API, we're migrating our backbone models to subclass keras_cv.models.Backbone.
This allows us to have a more structured, modular API for high-level task models, including the new ImageClassifier model (see #1458).

A great step-by-step guide to this migration can be found in the description of #1508, which performed this migration for ResNetV1.

If you'd like to take this issue up, please leave a comment and the KerasCV team can assign the issue to you. When opening a PR, please tag this issue as fixed by your PR, and tag our tracking bug #1627 for bookkeeping.

### PR DESCRIPTION
**Pull request by soma2000-lang**
# What does this PR do?

Fixes #1628

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [ ] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by jbischof — COMMENTED** [review]
I don't think ViT will work as a backbone preset due to the very different structure from convolution models. We may need to omit from `backbone_presets` and build a custom classifier. I don't think feature pyramids will be compatible either. (cc @ianstenbit)

**Comment by soma2000-lang** [comment]
@jbischof @jbischof  Should I omit the backbone presets then

File: keras_cv/models/backbones/vit/vit_backbone.py
Comment (jbischof) [inline keras_cv/models/backbones/vit/vit_backbone.py:1]: Good job using `git mv`, but you also need to rebase the code on the `Backbone` class.

File: keras_cv/models/backbones/vit/vit_backbone_presets.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+        "metadata": {
+            "description": (
+                "ViT model with 18 layers where the batch normalization "
+                "and ReLU activation are applied after the convolution layers "
+                "(v1 style)."
+            ),
+        },
+        "class_name": "keras_cv.models>ViTBackbone",
+        "config": {
+            "stackwise_filters": [64, 128, 256, 512],
```
Comment (jbischof) [inline keras_cv/models/backbones/vit/vit_backbone_presets.py:27]: This preset definitions still appear to be from the ResNet model. You can update once `ViTBackbone` is ready

**Review by jbischof — CHANGES_REQUESTED** [review]
Good start, now you need to rebase the class on `Backbone`

**Comment by soma2000-lang** [comment]
@jbischof will fo that

**Comment by jbischof** [comment]
Closing since we don't want to move forward on this for now

**Pull request closed without merging**